fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/MaterializedViewRewriter.java

```diff
@@ -1155,7 +1155,13 @@ private PredicateSplit getCompensationPredicates(RewriteContext rewriteContext,
         if (srcPu == null && targetPu != null) {
             // query: empid < 5
             // mv: empid < 5 or salary > 100
-            srcPu = Utils.compoundAnd(compensationEqualPredicate, compensationPr);
+            if (!isQueryAgainstView) {
+                // compensationEqualPredicate and compensationPr is based on query, need to change it to view based
+                srcPu = Utils.compoundAnd(columnRewriter.rewriteQueryToView(compensationEqualPredicate),
```

Comment:
`columnRewriter.rewriteQueryToView(compensationEqualPredicate)` can be null ?

Reply:
null is ok for compoundAnd